code/datums/elements/houlihan_teleport.dm

```diff
@@ -0,0 +1,86 @@
+/// Teleports interactors onto a safe turf randomly picked from a list of z-levels.
+/datum/element/houlihan_teleport
+	element_flags = ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	/// Text that will appear in the alert prompt.
+	var/question = "Travel back?"
+	/// List of z-levels that the user can teleport to. By default, this is station z-levels.
+	var/list/zlevels
+
+/datum/element/houlihan_teleport/Attach(datum/target, question, zlevels)
```

Comment:
zlevels being a list means the bespoke aspect of this will never actually do anything unless all input lists are static/deduped, and in doing that you're just adding extra complexity to users for no reason. If this is an ease of use feature we don't want that yea?

the reason to make this a component is it functionally already is one, and this will likely never be common enough code to justify making it not that. bespoke elements are a weak memory/setup optimization over components, and one that isn't really justified here. they only really exist for dumbass cases and this isn't that.